src/Build/TelemetryInfra/TelemetryDataUtils.cs

```diff
@@ -0,0 +1,320 @@
+                    TasksInfo customTaskInfo,
+                    TaskOrTargetTelemetryKey key,
+                    TaskExecutionStats taskExecutionStats)
+                {
+                    UpdateSingleStatistics(key.IsCustom ? customTaskInfo : builtinTaskInfo, taskExecutionStats, key);
+
+                    void UpdateSingleStatistics(TasksInfo summarizedTaskInfo, TaskExecutionStats infoToAdd, TaskOrTargetTelemetryKey kkey)
+                    {
+                        summarizedTaskInfo.Total.AddAnother(infoToAdd);
+                        if (kkey.IsFromNugetCache)
+                        {
+                            summarizedTaskInfo.FromNuget.AddAnother(infoToAdd);
+                        }
+                    }
+                }
+            }
+
+            private TasksInfo BuiltinTasksInfo { get; } = new TasksInfo();
+            private TasksInfo CustomTasksInfo { get; } = new TasksInfo();
+
+            private class TasksInfo
+            {
+                public TaskExecutionStats Total { get; } = TaskExecutionStats.CreateEmpty();
+                public TaskExecutionStats FromNuget { get; } = TaskExecutionStats.CreateEmpty();
+            }
+
+            public override TasksSummary? Read(
+                ref Utf8JsonReader reader,
+                Type typeToConvert,
+                JsonSerializerOptions options) =>
+            throw new NotImplementedException("Reading is not supported");
+
+            public override void Write(
+                Utf8JsonWriter writer,
+                TasksSummary value,
+                JsonSerializerOptions options)
+            {
+                writer.WriteStartObject();
+                WriteStat(writer, value.BuiltinTasksInfo, "Microsoft");
+                WriteStat(writer, value.CustomTasksInfo, "Custom");
+                writer.WriteEndObject();
+
+                void WriteStat(Utf8JsonWriter writer, TasksInfo tasksInfo, string name)
+                {
+                    writer.WriteStartObject(name);
+                    WriteSingleStat(writer, tasksInfo.Total, "Total", true);
+                    WriteSingleStat(writer, tasksInfo.FromNuget, "FromNuget", false);
+                    writer.WriteEndObject();
+                }
+
+                void WriteSingleStat(Utf8JsonWriter writer, TaskExecutionStats stats, string name, bool writeIfEmpty)
+                {
+                    if (stats.ExecutionsCount > 0)
+                    {
+                        writer.WriteStartObject(name);
+                        writer.WriteNumber("TotalExecutionsCount", stats.ExecutionsCount);
+                        // We do not want decimals
+                        writer.WriteNumber("CumulativeExecutionTimeMs", (long)stats.CumulativeExecutionTime.TotalMilliseconds);
+                        // We do not want decimals
+                        writer.WriteNumber("CumulativeConsumedMemoryKB", stats.TotalMemoryConsumption / 1024);
```

Comment:
can we use consistent naming for the variables and json keys, so it can be nameof(prop), instead on the hardcoded strings?